src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,56 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int DATA_CORRECT_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new FailedRegistrationException("User can't be null");
+        }
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        Storage.PEOPLE.add(user);
+        return user;
+    }
+
+    private void checkAge(User user) {
+        if (user.getAge() < MINIMUM_AGE) {
+            throw new FailedRegistrationException("Invalid age, "
+                    + "it must be at least " + MINIMUM_AGE);
+        }
+    }
+
+    private void checkPassword(User user) {
+        checkData(user.getPassword(), "Password");
+    }
+
+    private void checkLogin(User user) {
+        checkData(user.getLogin(), "Login");
```

Comment:
If you decide to check parameters in such way it's better to use constant for Login and Password 
```suggestion
        checkData(user.getLogin(), TYPE_LOGIN);
```